src/transformers/models/llama/convert_llama_weights_to_hf.py

```diff
@@ -120,11 +126,15 @@ def permute(w):
             }
         else:
             # Sharded
+            # Note that in the 13B checkpoint, not cloning the two following weights will result in the checkpoint
+            # becoming 37GB instead of 26GB for some reason.
             state_dict = {
-                f"model.layers.{layer_i}.input_layernorm.weight": loaded[0][f"layers.{layer_i}.attention_norm.weight"],
+                f"model.layers.{layer_i}.input_layernorm.weight": loaded[0][
+                    f"layers.{layer_i}.attention_norm.weight"
+                ].clone(),
```

Comment:
Debugged this part and for some reason, those weights once saved take an absurd amount memory (more than 110Mb instead of 5kb) without this clone. I think it's because the tensors keep references to "something else". The clone() here and below fixes the issue.

Reply:
@ArthurZucker @LysandreJik  just curious, do we have any idea how this could happen? is it sharing weights or something